Extend and improve OCI terraform README (#101)
lacework-dev/scripts: lw-oci-integration/tf/README.md

    fingerprint="<YOUR KEY FINGER PRINT>"
    private_key_path="<YOUR KEY KEY PATH (e.g. ~/.oci/myocikey.pem)>"
    ```
4. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).

## Prepare to run Terraform via OCI Cloud Shell (Option 2)

1. Open the OCI cloud shell. For example:
   1. Log in to the OCI console.
   2. Click on the '<>' icon (developer tools), located on the top right of the OCI web console.
   3. Select Cloud Shell.
2. Clone this github repo. For example:
    ```
    git clone https://github.com/lacework-dev/scripts.git
    ```
3. Change directory to the `lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.
4. OCI auth is automatically handled by the cloud shell. To use this, we modify a couple of files:
   * In `main.tf`, edit the provider section so that it looks like this (setting `tenancy_ocid` and `region` accordingly):
     
     ```
     provider "oci" {
       tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaatxph3zpu3jkem4cseejktwllb6dgsam7cx3jy63pazejzobvjqqq"
       region = "us-sanjose-1"
     } 
     

>      ```

Reviewer: Hmmm, actually the `main.tf` today uses the variables from tfvars to set the provider details:
```
provider "oci" {
  tenancy_ocid     = var.tenancy_ocid
  user_ocid        = var.user_ocid
  fingerprint      = var.fingerprint
  private_key_path = var.private_key_path
  region           = var.region
}
```

I don't think we need to or should recommend hard coding these values.

Author: Thanks - I see main.tf change has been reverted and instead the instruction for cloudshell is to set non-required variables to null:
```
     user_ocid=null
     fingerprint=null
     private_key_path=null
```

## Lacework OCI Integration

> You can run the Terraform scripts locally or via an OCI cloud shell. We recommend using the OCI cloud shell, since it comes with Terraform pre-installed and provides for authentication.

Reviewer 2: ```suggestion
The OCI Terraform scripts can be run locally or via an OCI cloud shell. Using the OCI cloud shell is the recommended option, as it comes with Terraform pre-installed and aids in authentication.
```

Reviewer 3: ```suggestion
The OCI Terraform scripts can be run locally or through an OCI cloud shell. Using the OCI cloud shell is recommended, as it comes with Terraform pre-installed and helps with authentication.
```

## Prepare to run Terraform locally (Option 1)

Before using the Terraform script, you must create the required environment variables to access your OCI tenant. The recommended way to do this is:

> 1. Clone this github repo. For example:

Reviewer 2: Should we have a pre-reqs section here to have git & Terraform installed locally?
Also what is the min version of TF that this has been tested with so far that we could call out as the min supported version?

   ```
   git clone https://github.com/lacework-dev/scripts.git
   ```
2. Change directory to the `lw-oci-integration/tf` folder and copy the `vars.tfvars.example` to `terraform.tfvars`.
3. Update the values in this file to align with your administrative OCI user access.
    ```
    region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
    tenancy_ocid="<YOUR OCI TEANCY OCID>"
    user_ocid="<YOUR OCI USER OCID>"
    fingerprint="<YOUR KEY FINGER PRINT>"
    private_key_path="<YOUR KEY KEY PATH (e.g. ~/.oci/myocikey.pem)>"
    ```

> 4. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).

Reviewer 2: "(if it is not already installed)" seems unnecessary IMO
```suggestion
4. Install and configure the Lacework CLI with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```

Reviewer 2: Second suggestion
```suggestion
4. Install the Lacework CLI and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).
```

Reviewer 3: Yes that's fine. Users will ignore the advice to install the CLI if they already have it. No need to call it explicitly.

## Prepare to run Terraform via OCI Cloud Shell (Option 2)

1. Open the OCI cloud shell. For example:
   1. Log in to the OCI console.
   2. Click on the '<>' icon (developer tools), located on the top right of the OCI web console.
   3. Select Cloud Shell.
2. Clone this github repo. For example:
    ```
    git clone https://github.com/lacework-dev/scripts.git
    ```

> 3. Change directory to the `lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.

Reviewer 2: ```suggestion
3. Change directory to the `scripts/lw-oci-integration/tf` folder and copy `vars.tfvars.example` to `terraform.tfvars`.
    ```
    cd scripts/lw-oci-integration/tf
   cp  vars.tfvars.example terraform.tfvars
    ```
```

> 4. OCI auth is automatically handled by the cloud shell. To use this, we modify a couple of files:

Reviewer 2: ```suggestion
4. OCI auth is automatically handled by the cloud shell. To use this, modify the following files:
```

Reviewer 2: Actually, it looks like we only edit `terraform.tfvars` in which case my suggestion would be
```suggestion
4. OCI auth is automatically handled by the cloud shell. To use this, modify the values in `terraform.tfvars`:
```

Reviewer 3: ```suggestion
4. OCI authentication is automatically handled by the cloud shell. To use this, modify the values in `terraform.tfvars`:
```

>    * In `terraform.tfvars`, set the `region` and `tenancy_ocid` as appropriate. The `user_ocid`, `fingerprint` 
> and `private_key_path` fields are not applicable when using cloud shell. The `group_name`, `user_name` and `policy_name` will be
> created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.

Reviewer 2: ```suggestion
   * `region` and `tenancy_ocid` should be set as appropriate.
   * `user_ocid`, `fingerprint` and `private_key_path` fields are not applicable when using cloud shell and should be set to null.
   * `group_name`, `user_name` and `policy_name` will be
created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.
```

Reviewer 3: ```suggestion
   * `region` and `tenancy_ocid` should be set as appropriate.
	   * `user_ocid`, `fingerprint` and `private_key_path` fields are not applicable when using cloud shell and should be set to null.
	   * `group_name`, `user_name` and `policy_name` will be created by the Terraform script for the Lacework OCI integration. They can be left as they are, or modified if required.
```

     
     ```
     region="<YOUR HOME REGION  (e.g. us-ashburn-1)>"
     tenancy_ocid="<YOUR OCI TEANCY OCID>"
     user_ocid=null
     fingerprint=null
     private_key_path=null
     
     ```

> 5. Install the Lacework CLI (if it is not already installed) and configure with your API key. For more information, see the [Lacework CLI Guide](https://docs.lacework.net/cli/).

Reviewer 2: Same suggestion as above

     
   **Note:** The step to install the CLI may fail with permission denied on the OCI cloud shell. If this happens, follow these steps:
   1. Download the install script
      ```
      curl https://raw.githubusercontent.com/lacework/go-sdk/main/cli/install.sh
      ```
   2. Install to the present working directory (PWD):
      ```
      ./install.sh -d $PWD
      ```
   3. Modify the path (temporary) to include the PWD:
      ```
      export PATH=$PATH:$PWD
      ```

> Now, continue following instructions on [Lacework CLI Guide](https://docs.lacework.net/cli/) to configure your API key.

Reviewer 2: This is already linked in the text for step 5, should we remove this link?

> # Running Terraform (common to options 1 and option 2)

Reviewer 2: IMO it is implied that it is common
```suggestion
# Running Terraform
```

Reviewer 3: Yes that's usually better practice.

> 1. Once the file is updated, run the following command to initialize Terraform:

Reviewer 2: Which file?

Author: Remove "Once the file is updated,"

   ```
   terraform init
   ```

> 2. Now verify and plan the Terraform using this command:

Reviewer 2: ```suggestion
2. Now verify and generate a Terraform plan:
```

   ```
   terraform plan
   ```

> 3. If everything looks good, apply the change using the following command:

Reviewer 2: Again, this is slightly vague... Maybe something along the lines of:
```suggestion
3. If `Terraform Plan` runs successfully with no errors, lets run the Terraform to create the required OCI resources:
```

Reviewer 3: ```suggestion
3. If `Terraform Plan` runs successfully with no errors, use the following command to create the required OCI resources:
```

   ```
   terraform apply -auto-approve
   ```

> 4. After the plan is applied, a Lacework cloud account policy is created locally under ~/.oci/lacework_cloud_account.json. You can post this file to Lacework via the Lacework CLI to create your integration: 

Reviewer 2: ```suggestion
4. After the the Terraform is applied, a Lacework cloud account payload is created locally under ~/.oci/lacework_cloud_account.json. In order to create your OCI integration on the Lacework platform, run the following post: 
```

Reviewer 3: ```suggestion
4. After the Terraform is applied, a Lacework cloud account payload is created locally under `~/.oci/lacework_cloud_account.json`. To create your OCI integration on the Lacework platform, run the following post:
```
